clusterversion: rename latest/min supported versions in Handle

RaduBerinde opened the pull request:
Rename "binary version" to "latest version" and "binary min supported
version" to "min supported version" in `clusterversion.Handle` and
associated code.

This change is mechanical and is not intended to have any functional changes.

Informs: #112629
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/113727)
<!-- Reviewable:end -->

andyyang890 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 36 of 36 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/113727#-:-NiLsLrmBeNrQpXhsdEe:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian, @dt, @itsbilal, and @RaduBerinde)

___
*[`pkg/clusterversion/clusterversion.go` line 14 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/113727#-NiLo98zDZBJ8J5vCbvv:-NiLo98zDZBJ8J5vCbvw:b6k3p4l) ([raw file](https://github.com/cockroachdb/cockroach/blob/0dcacbd02064598264205c2498a41ba6c9787cc1/pkg/clusterversion/clusterversion.go#L14)):*
> ```Go
> // versions in order to accommodate backward incompatible behaviors. It handles
> // the feature gates and so must maintain a fairly lightweight set of
> // dependencies. The upgrade sub-package handles advancing a cluster from one
> ```

nit: s/sub-package/package/


<!-- Sent from Reviewable.io -->

RaduBerinde committed:
clusterversion: rename latest/min supported versions in Handle

Rename "binary version" to "latest version" and "binary min supported
version" to "min supported version" in `clusterversion.Handle` and
associated code.

Informs: #112629
Release note: None

RaduBerinde reviewed (commented):
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/113727#-:-NiLuVaQAf8UFtDuzlAv:b-814g8k)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian, @dt, and @itsbilal)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
bors r+
